### PR TITLE
LOG-5903: fix kafka topic being required

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* fixes the any of the collection e2e tests not being run
* makes kafka topic optional

### Links
https://issues.redhat.com/browse/LOG-5903

cc @vparfonov @cahartma @Clee2691

### CODE REVIEW

File: api/observability/v1/output_types.go [hunk 1]
```diff
@@ -601,6 +601,7 @@ type Kafka struct {
 	// 3. foo.{.bar.baz||.qux.quux.corge||.grault||"nil"}-waldo.fred{.plugh||"none"}
 	//
 	// +kubebuilder:validation:Pattern:=`^(([a-zA-Z0-9-_.\/])*(\{(\.[a-zA-Z0-9_]+|\."[^"]+")+((\|\|)(\.[a-zA-Z0-9_]+|\.?"[^"]+")+)*\|\|"[^"]*"\})*)*$`
+	// +kubebuilder:validation:Optional
 	Topic string `json:"topic"`
```
Comment (Clee2691) [inline api/observability/v1/output_types.go:605]: should add `omitempty` here

File: api/observability/v1/output_types.go [hunk 2]
```diff
@@ -601,7 +601,8 @@ type Kafka struct {
 	// 3. foo.{.bar.baz||.qux.quux.corge||.grault||"nil"}-waldo.fred{.plugh||"none"}
 	//
 	// +kubebuilder:validation:Pattern:=`^(([a-zA-Z0-9-_.\/])*(\{(\.[a-zA-Z0-9_]+|\."[^"]+")+((\|\|)(\.[a-zA-Z0-9_]+|\.?"[^"]+")+)*\|\|"[^"]*"\})*)*$`
-	Topic string `json:"topic"`
+	// +kubebuilder:validation:Optional
```
Comment (cahartma) [inline api/observability/v1/output_types.go:604]: My understanding is you need `omitempty` here according to Alan's explanation?

Comment (jcantrill) [reply api/observability/v1/output_types.go:604]: added per suggestion of calvin... see line 605

**Pull request merged**